src/utils/record-completeness/ComputeRecordCompleteness.ts

```diff
@@ -1,92 +1,113 @@
+    return RECORD_COMPLETENESS_ENUM.SKELETON;
+  }
+
+  const mandatoryAttributesValidationResult = validateMandatoryTechRecordAttributes(vehicle);
+
+  const isCoreMandatoryValid = !mandatoryAttributesValidationResult.coreValidationResult?.error;
+
+  const isNonCoreMandatoryValid = !mandatoryAttributesValidationResult.nonCoreValidationResult?.error;
+
+  if (isCoreMandatoryValid && isNonCoreMandatoryValid) {
+    return RECORD_COMPLETENESS_ENUM.COMPLETE;
+  } else if (isCoreMandatoryValid) {
+    return RECORD_COMPLETENESS_ENUM.TESTABLE;
   } else {
-    return validateRecordCompleteness(coreMandatoryValidation.trlCoreMandatoryVehicleAttributes, vehicleAttributes);
+    return RECORD_COMPLETENESS_ENUM.SKELETON;
   }
 };
 
-const validateCoreAndNonCoreMandatoryTechRecordAttributes = (vehicleType: string, techRecordWrapper: Vehicle) => {
-  let coreMandatoryValidationResult;
-  let nonCoreMandatoryValidationResult;
-  let coreMandatorySchema: ObjectSchema;
-  let nonCoreMandatorySchema: ObjectSchema | undefined;
-  let techRecord = techRecordWrapper.techRecord[0];
-  if (vehicleType === VEHICLE_TYPE.HGV) {
-    coreMandatorySchema = coreMandatoryValidation.hgvCoreMandatorySchema;
-    nonCoreMandatorySchema = nonCoreMandatoryValidation.hgvNonCoreMandatorySchema;
-  } else if (vehicleType === VEHICLE_TYPE.PSV) {
-    coreMandatorySchema = coreMandatoryValidation.psvCoreMandatorySchema;
-    nonCoreMandatorySchema = nonCoreMandatoryValidation.psvNonCoreMandatorySchema;
-  } else if (vehicleType === VEHICLE_TYPE.TRL) {
-    techRecord = {...techRecord, primaryVrm: techRecordWrapper.primaryVrm} as any;
-    coreMandatorySchema = coreMandatoryValidation.trlCoreMandatorySchema;
-    nonCoreMandatorySchema = nonCoreMandatoryValidation.trlNonCoreMandatorySchema;
-  } else if (vehicleType === VEHICLE_TYPE.LGV) {
-    coreMandatorySchema = coreMandatoryValidation.lgvCoreMandatorySchema;
-  } else if (vehicleType === VEHICLE_TYPE.CAR) {
-    coreMandatorySchema = coreMandatoryValidation.carCoreMandatorySchema;
-  } else if (vehicleType === VEHICLE_TYPE.MOTORCYCLE) {
-    coreMandatorySchema = coreMandatoryValidation.motorcycleCoreMandatorySchema;
+function validateVehicleAttributes(vehicle: Vehicle): ValidationResult | undefined {
+  const vehicleType = vehicle.techRecord[0].vehicleType;
+
+  const validationSchema = vehicleType === VEHICLE_TYPE.TRL
+    ? trlCoreMandatoryVehicleAttributes
+    : psvHgvCarLgvMotoCoreMandatoryVehicleAttributes
+
+  if (!vehicleType) {
+    return missingVehicleTypeValidationResult;
+  } else if (!validationSchema) {
+    return undefined;
   } else {
-    throw new HTTPError(400, ERRORS.VEHICLE_TYPE_ERROR);
+    const vehicleAttributes = {
+      systemNumber: vehicle.systemNumber,
+      vin: vehicle.vin,
+      primaryVrm: vehicle.primaryVrm,
+      trailerId: (vehicle as Trailer).trailerId ?? undefined
+    };
+
+    return validationSchema.validate(vehicleAttributes, { stripUnknown: true });
```

Comment:
```suggestion
    return validationSchema?.validate(vehicleAttributes, { stripUnknown: true });
```

I _believe_ adding this would negate the need for the `else if` on lines 53  and 54.

Reply:
also Aaron's comment

Reply:
Wow, that's clean. Suggestion applied.